tests/src/test/java/com/datastax/oss/pulsar/functions/transforms/tests/AbstractDockerTest.java

```diff
@@ -389,6 +403,45 @@ public void testComputeAvroDateToString() throws Exception {
         "{\"dateField\": \"2023-04-01\", \"optionalDateField\": \"2023-04-02\"}");
   }
 
+  /**
+   * TODO: Enable non-optional decimal tests. For now the tests works fine with optional only or non-optional. However,
+   * enabling both results in org.apache.pulsar.client.api.SchemaSerializationException: java.io.EOFException
+   */
+  @Test
+  public void testComputeCqlDecimalToAvroDecimal() throws Exception {
+    String userConfig =
+        ("{\"steps\": [{\"type\": \"compute\", \"fields\": ["
+            //+ "{\"name\": \"value.cqlDecimal\", \"expression\": \"fn:decimal(value.cqlDecimal.bigint, value.cqlDecimal.scale)\", \"type\": \"DECIMAL\"},"
```

Comment:
Yeas I'll probably delete, or re-enable. If you check the TODO message above.

I was not sure why if I have both optional and non-optional decimal, publishing to the data topic would fail. I'll troubleshoot more. Perhaps, I can create to separate tests. The reason I wanted to test both is because the CQL Decimal will be non-optional in the key part (primary key, assuming it originated from C* CDC) but optional in the value.

Reply:
I ended up using a KeyValue schema with non optional decimal in key, and optional in value and it went through (also it is closer to what CDC would do). The fact the both cannot coexist in the value remains a puzzle to me (seems logic in Avro and not pulsar client that would cause `java.io.EOFException`)